static/src/javascripts/projects/common/modules/gallery/lightbox.js

```diff
@@ -0,0 +1,626 @@
+                }
+            },
+            prev() {
+                pulseButton(this.prevBtn);
+                if (this.index === 1) {
+                    // first img
+                    if (this.showEndslate) {
+                        this.state = 'endslate';
+                    } else {
+                        this.index = this.images.length;
+                        this.reloadState = true;
+                    }
+                } else {
+                    this.index -= 1;
+                    this.reloadState = true;
+                }
+            },
+            reload() {
+                this.reloadState = true;
+            },
+            toggleInfo() {
+                pulseButton(this.infoBtn);
+                this.$lightboxEl.toggleClass('gallery-lightbox--show-info');
+            },
+            hideInfo() {
+                pulseButton(this.infoBtn);
+                this.$lightboxEl.removeClass('gallery-lightbox--show-info');
+            },
+            showInfo() {
+                pulseButton(this.infoBtn);
+                this.$lightboxEl.addClass('gallery-lightbox--show-info');
+            },
+            resize() {
+                this.swipeContainerWidth = this.$swipeContainer.dim().width;
+                this.loadSurroundingImages(this.index, this.images.length); // regenerate src
+                this.translateContent(this.index, 0, 0);
+            },
+            close() {
+                this.state = 'closed';
+            },
+        },
+    },
+
+    endslate: {
+        enter() {
+            this.translateContent(this.$slides.length, 0, 0);
+            this.index = this.images.length + 1;
+            mediator.on('window:throttledResize', this.resize);
+        },
+        leave() {
+            mediator.off('window:throttledResize', this.resize);
+        },
+        events: {
+            next() {
+                pulseButton(this.nextBtn);
+                this.index = 1;
+                this.state = 'image';
+            },
+            prev() {
+                pulseButton(this.prevBtn);
+                this.index = this.images.length;
+                this.state = 'image';
+            },
+            reload() {
+                this.reloadState = true;
+            },
+            resize() {
+                this.swipeContainerWidth = this.$swipeContainer.dim().width;
+                this.translateContent(this.$slides.length, 0, 0);
+            },
+            close() {
+                this.state = 'closed';
+            },
+        },
+    },
+};
+
+GalleryLightbox.prototype.endslate = new Component();
+
+const init = (): void => {
+    loadCssPromise.then(() => {
+        const images = config.get('page.lightboxImages');
+
+        if (images && images.images.length > 0) {
+            let lightbox;
+            const galleryHash = window.location.hash;
+
+            let res;
+
+            bean.on(document.body, 'click', '.js-gallerythumbs', e => {
```

Comment:
Because the callback is so big: could you add the `:Event` type definition here?